src/target/stm32h7.c

```diff
@@ -343,25 +371,55 @@ static bool stm32h7_flash_unlock(target_s *const target, const uint32_t addr)
 	return !(target_mem_read32(target, regbase + FLASH_CR) & FLASH_CR_LOCK);
 }
 
+/* Helper for offsetting FLASH_CR bits correctly */
+static uint32_t stm32h7_flash_cr(uint32_t sector_size, const uint32_t ctrl, int snb)
```

Comment:
Ref.mans state either SNB or SSN for Sector Selection Number bitfield FLASH_CR. Inside BMD driver it's a small unsigned integer 0..7 (0..127) depending on exact part. Previously it was multiplied by some offset bit to land in place, and I didn't feel like introducing more macros; but if that's more readable, then please suggest an alternative to `FLASH_CR_SNB_1` counterpart for H7Bx. I can make it `uint32_t snb` or `unsigned int` or whatever, it can be `const`, the caller doesn't see this, the callee doesn't modify it.

Reply:
Please go with `const uint8_t sector_number` then - this should avoid all the unpleasantness and gives a more meaningful name. The shift below is not the problem, the cryptic naming is, along with the bad type choice.

Reply:
Having been poking this a bit ourself for the H723/H733, we additionally suggest replacing `FLASH_CR_SNB_1` with `FLASH_CR_SNB_SHIFT` which gets defined as 8U, and turning the multiplication into a right shift. It makes the whole lot make a bit more sense we think.

Reply:
Renamed argument and introduced macros.